File: Ahorn/entities/DashCopyBerry.jl
module DashCopyBerry

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/DashCopyBerry" DashCopyBerry(x::Integer, y::Integer, dashRefill::Integer=1, power::String="Sword", refillOnlyWhenEmpty::Bool=false)

const placements = Ahorn.PlacementDict(
    "dash_copy_berry" => Ahorn.EntityPlacement(DashCopyBerry),
    "dash_copy_berry_fire" => Ahorn.EntityPlacement(DashCopyBerry),
    "dash_copy_berry_wing" => Ahorn.EntityPlacement(DashCopyBerry)
)

function Ahorn.selection(entity::DashCopyBerry)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::DashCopyBerry, room::Maple.Room)
    Ahorn.drawSprite(ctx, "collectables/strawberry/normal00", entity.x, entity.y)
end

end

File: framework/extracted/Loenn/entities/DashCopyBerry.lua
local dashCopyBerry = {}

dashCopyBerry.name = "MaggyHelper/DashCopyBerry"
dashCopyBerry.depth = -100
dashCopyBerry.texture = "collectables/strawberry/normal00"

dashCopyBerry.fieldInformation = {
    power = {
        fieldType = "string",
        options = {
            "None", "Fire", "Ice", "Spark", "Sword", "Cutter", "Beam",
            "Stone", "Needle", "Parasol", "Wheel", "Bomb", "Fighter",
            "Suplex", "Ninja", "Mirror", "Hammer", "Knight", "Wing",
            "UFO", "Sleep"
        },
        editable = false
    },
    dashRefill = {
        fieldType = "integer",
        minimumValue = 1,
        maximumValue = 5
    }
}

dashCopyBerry.placements = {
    {
        name = "dash_copy_berry",
        data = {
            power = "Sword",
            dashRefill = 1,
            refillOnlyWhenEmpty = false
        }
    },
    {
        name = "dash_copy_berry_fire",
        data = {
            power = "Fire",
            dashRefill = 1,
            refillOnlyWhenEmpty = false
        }
    },
    {
        name = "dash_copy_berry_wing",
        data = {
            power = "Wing",
            dashRefill = 2,
            refillOnlyWhenEmpty = false
        }
    }
}

return dashCopyBerry
